server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -174,7 +175,7 @@ private void assertSystemMailboxesAreNotUpdated(Mailbox mailbox, MailboxUpdateRe
     @VisibleForTesting
     <T> boolean requestChanged(Optional<T> requestValue, Optional<T> storeValue) {
         return requestValue
-            .filter(value -> !requestValue.equals(storeValue))
+            .filter(Predicate.not(Predicate.isEqual(storeValue)))
```

Comment:
I don't see how this syntax can work (probably the root cause as @chibenwa said) with two predicates

We want it not equal for filtering right? It's gonna check first the `Predicate.isEqual`. If it's equal, then it's gonna go to the not predicate, which will fail. If it's not equal, it fails this predicate. My guess is you probably filter out everytime with this syntax? (same remark for all blocks using this)

Reply:
Predicate.not() is just inversing Predicate.isEqual. So if Predicate.IsEqual is false, not(isEqual) will be true